diff --git a/targets/TARGET_STM/TARGET_STM32F3/STM32Cube_FW/STM32F3xx_HAL_Driver/stm32f3xx_ll_rtc.c b/targets/TARGET_STM/TARGET_STM32F3/STM32Cube_FW/STM32F3xx_HAL_Driver/stm32f3xx_ll_rtc.c
index ef20bbc1..5f1c2527 100644
--- a/targets/TARGET_STM/TARGET_STM32F3/STM32Cube_FW/STM32F3xx_HAL_Driver/stm32f3xx_ll_rtc.c
+++ b/targets/TARGET_STM/TARGET_STM32F3/STM32Cube_FW/STM32F3xx_HAL_Driver/stm32f3xx_ll_rtc.c
@@ -776,17 +776,36 @@ ErrorStatus LL_RTC_EnterInitMode(RTC_TypeDef *RTCx)
   * @param  RTCx RTC Instance
   * @retval An ErrorStatus enumeration value:
   *          - SUCCESS: RTC exited from in Init mode
-  *          - ERROR: Not applicable
+  *          - ERROR: RTC failed to exit Init mode
   */
 ErrorStatus LL_RTC_ExitInitMode(RTC_TypeDef *RTCx)
 {
+  __IO uint32_t timeout = RTC_INITMODE_TIMEOUT;
+  ErrorStatus status = SUCCESS;
+  uint32_t tmp = 0U;
+
   /* Check the parameter */
   assert_param(IS_RTC_ALL_INSTANCE(RTCx));
 
   /* Disable initialization mode */
   LL_RTC_DisableInitMode(RTCx);
 
-  return SUCCESS;
+  /* Wait until the INITF flag is cleared */
+  tmp = LL_RTC_IsActiveFlag_INIT(RTCx);
+  while ((timeout != 0U) && (tmp != 0U))
+  {
+    if (LL_SYSTICK_IsActiveCounterFlag() == 1U)
+    {
+      timeout--;
+    }
+    tmp = LL_RTC_IsActiveFlag_INIT(RTCx);
+    if (timeout == 0U)
+    {
+      status = ERROR;
+    }
+  }
+
+  return status;
 }
 
 /**
