diff --git a/targets/TARGET_Cypress/TARGET_PSOC6/mtb-pdl-cat1/drivers/include/cy_pdm_pcm.h b/targets/TARGET_Cypress/TARGET_PSOC6/mtb-pdl-cat1/drivers/include/cy_pdm_pcm.h
index 718a031f..379b64ce 100644
--- a/targets/TARGET_Cypress/TARGET_PSOC6/mtb-pdl-cat1/drivers/include/cy_pdm_pcm.h
+++ b/targets/TARGET_Cypress/TARGET_PSOC6/mtb-pdl-cat1/drivers/include/cy_pdm_pcm.h
@@ -618,7 +618,7 @@ __STATIC_INLINE uint32_t Cy_PDM_PCM_GetInterruptStatus(PDM_Type const * base)
 * Function Name: Cy_PDM_PCM_ClearInterrupt
 ***************************************************************************//**
 *
-* Clears one or more PDM-PCM interrupt statuses (sets an INTR register's bits).
+* Clears one or more PDM-PCM interrupt statuses by writing to the INTR register.
 *
 * \param base The pointer to the PDM-PCM instance address
 * \param interrupt
