diff --git a/openair2/RRC/LTE/rrc_eNB_M2AP.c b/openair2/RRC/LTE/rrc_eNB_M2AP.c
index 63760907fd..ed6e560744 100644
--- a/openair2/RRC/LTE/rrc_eNB_M2AP.c
+++ b/openair2/RRC/LTE/rrc_eNB_M2AP.c
@@ -870,9 +870,12 @@ static uint8_t rrc_M2AP_do_SIB23_SIB2(
 
   rrc_mac_config_req_eNB_t tmp = {0};
   tmp.CC_id = CC_id;
+  tmp.rnti = 0xfffd;
   tmp.mbsfn_SubframeConfigList = carrier->sib2->mbsfn_SubframeConfigList;
   tmp.MBMS_Flag = carrier->MBMS_flag;
-  rrc_mac_config_req_eNB(ctxt_pP->module_id, &tmp);
+  tmp.mbsfn_AreaInfoList = &carrier->sib13->mbsfn_AreaInfoList_r9;
+  tmp.FeMBMS_Flag = carrier->FeMBMS_flag;
+  rrc_mac_config_req_eNB(Mod_id, &tmp);
 
   RC.rrc[ctxt_pP->module_id]->carrier[CC_id].sizeof_SIB23 = ((enc_rval.encoded+7)/8);
  
@@ -1268,10 +1271,12 @@ static uint8_t rrc_M2AP_do_SIB23_SIB2_SIB13(
 
   rrc_mac_config_req_eNB_t tmp = {0};
   tmp.CC_id = CC_id;
+  tmp.rnti = 0xfffd;
   tmp.mbsfn_SubframeConfigList = carrier->sib2->mbsfn_SubframeConfigList;
   tmp.MBMS_Flag = carrier->MBMS_flag;
   tmp.mbsfn_AreaInfoList = &carrier->sib13->mbsfn_AreaInfoList_r9;
-  rrc_mac_config_req_eNB(ctxt_pP->module_id, &tmp);
+  tmp.FeMBMS_Flag = carrier->FeMBMS_flag;
+  rrc_mac_config_req_eNB(Mod_id, &tmp);
 
   RC.rrc[ctxt_pP->module_id]->carrier[CC_id].sizeof_SIB23 = ((enc_rval.encoded+7)/8);
  
