diff --git a/openair1/PHY/LTE_ESTIMATION/lte_eNB_measurements.c b/openair1/PHY/LTE_ESTIMATION/lte_eNB_measurements.c
index f1b26c8d4e..2c38f0a994 100644
--- a/openair1/PHY/LTE_ESTIMATION/lte_eNB_measurements.c
+++ b/openair1/PHY/LTE_ESTIMATION/lte_eNB_measurements.c
@@ -97,7 +97,6 @@ void lte_eNB_I0_measurements(PHY_VARS_eNB *eNB,
     measurements->n0_power_tot_dBm = measurements->n0_power_tot_dB - eNB->rx_total_gain_dB;
   //      printf("n0_power %d\n",measurements->n0_power_tot_dB);
 
-  }
 
   n0_power_tot2=0;
   int nb_rb=0;
