diff --git a/openair1/PHY/NR_REFSIG/ptrs_nr.c b/openair1/PHY/NR_REFSIG/ptrs_nr.c
index f73a01a2d6..bc3afaeacb 100644
--- a/openair1/PHY/NR_REFSIG/ptrs_nr.c
+++ b/openair1/PHY/NR_REFSIG/ptrs_nr.c
@@ -356,10 +356,9 @@ void get_slope_from_estimates(uint8_t start, uint8_t end, int16_t *est_p, double
 /* estimate from slope */
 void ptrs_estimate_from_slope(int16_t *error_est, double *slope_p, uint8_t start, uint8_t end)
 {
-  c16_t *error=(struct complex16 *) error_est;
   for(uint8_t i = 1; i< (end -start);i++) {
-    error[start+i].r = error[start].r + (int16_t)(i * slope_p[0]);// real
-    error[start+i].i = error[start].i + (int16_t)(i * slope_p[1]); //imag
+    error_est[(start+i)*2] = error_est[start*2] + (int16_t)(i * slope_p[0]);// real
+    error_est[((start+i)*2)+1] = error_est[(start*2)+1] + (int16_t)(i * slope_p[1]); //imag
 #ifdef DEBUG_PTRS
     printf("[PHY][PTRS]: Estimated Symbol %2d -> %4d %4d from Slope (%4f %4f)\n", start+i,error_est[(start+i)*2],error_est[((start +i)*2)+1],
            slope_p[0],slope_p[1]);
