diff --git a/openair1/PHY/CODING/3gpplte_turbo_decoder.c b/openair1/PHY/CODING/3gpplte_turbo_decoder.c
index 52860e096f..9b7a050ec2 100644
--- a/openair1/PHY/CODING/3gpplte_turbo_decoder.c
+++ b/openair1/PHY/CODING/3gpplte_turbo_decoder.c
@@ -895,7 +895,7 @@ unsigned char phy_threegpplte_turbo_decoder_scalar(llr_t *y,
   /*  y is a pointer to the input
     decoded_bytes is a pointer to the decoded output
     n is the size in bits of the coded block, with the tail */
-  short ext[n],ext2[n];
+  short ext[n+3],ext2[n+3];
 
   //  short systematic0[n],systematic1[n],systematic2[n],yparity1[n],yparity2[n];
   llr_t *yp = y;
