diff --git a/openair2/RRC/NR/rrc_gNB_radio_bearers.c b/openair2/RRC/NR/rrc_gNB_radio_bearers.c
index 205a58fc36..60f52d47b4 100644
--- a/openair2/RRC/NR/rrc_gNB_radio_bearers.c
+++ b/openair2/RRC/NR/rrc_gNB_radio_bearers.c
@@ -111,6 +111,9 @@ drb_t *generateDRB(gNB_RRC_UE_t *ue,
     est_drb->cnAssociation.sdap_config.sdap_HeaderDL = NR_SDAP_Config__sdap_HeaderDL_absent;
     est_drb->cnAssociation.sdap_config.sdap_HeaderUL = NR_SDAP_Config__sdap_HeaderUL_absent;
   }
+  /* Clear the mappedQoS_FlowsToAdd array first to remove any stale values */
+  memset(est_drb->cnAssociation.sdap_config.mappedQoS_FlowsToAdd, 0, sizeof(est_drb->cnAssociation.sdap_config.mappedQoS_FlowsToAdd));
+  
   for (int qos_flow_index = 0; qos_flow_index < pduSession->param.nb_qos; qos_flow_index++) {
     est_drb->cnAssociation.sdap_config.mappedQoS_FlowsToAdd[qos_flow_index] = pduSession->param.qos[qos_flow_index].qfi;
     if (pduSession->param.qos[qos_flow_index].fiveQI > 5)
