diff --git a/openair2/LAYER2/nr_pdcp/nr_pdcp_oai_api.c b/openair2/LAYER2/nr_pdcp/nr_pdcp_oai_api.c
index a416ddd617..0b5b213c86 100644
--- a/openair2/LAYER2/nr_pdcp/nr_pdcp_oai_api.c
+++ b/openair2/LAYER2/nr_pdcp/nr_pdcp_oai_api.c
@@ -805,7 +805,7 @@ void add_srb(int is_gnb,
                                   deliver_sdu_srb,
                                   ue,
                                   NULL,
-                                  ue,
+                                  NULL,   // deliver_pdu_data not used for SRBs
                                   SHORT_SN_SIZE,
                                   t_Reordering,
                                   -1,
