diff --git a/common/utils/telnetsrv/telnetsrv_enb_measurements.c b/common/utils/telnetsrv/telnetsrv_enb_measurements.c
index b8898b665e..5e03cc1035 100644
--- a/common/utils/telnetsrv/telnetsrv_enb_measurements.c
+++ b/common/utils/telnetsrv/telnetsrv_enb_measurements.c
@@ -145,7 +145,7 @@ void measurcmd_display_one_rlcstat(telnet_printfunc_t prnt, int UE_id, telnet_lt
                                 rlcstats+6, rlcstats+7, rlcstats+8, rlcstats+9, rlcstats+10, rlcstats+11,
                                 rlcstats+12, rlcstats+13, rlcstats+14, rlcstats+15, rlcstats+16, rlcstats+17,
                                 rlcstats+18, rlcstats+19, rlcstats+20, rlcstats+21, rlcstats+22, rlcstats+23,
-                                rlcstats+24, rlcstats+25, rlcstats+26, rlcstats+27);
+                                rlcstats+24, rlcstats+25, rlcstats+26, rlcstats+27, rlcstats+28, rlcstats+29);
 
   if (rlc_status == RLC_OP_STATUS_OK) {
     prnt("%s UE %i RLC %s mode %s %s\n",HDR,UE_id, rbid_str,
