diff --git a/openair1/PHY/LTE_UE_TRANSPORT/linear_preprocessing_rec.c b/openair1/PHY/LTE_UE_TRANSPORT/linear_preprocessing_rec.c
index 9ac775c4be..4724285925 100644
--- a/openair1/PHY/LTE_UE_TRANSPORT/linear_preprocessing_rec.c
+++ b/openair1/PHY/LTE_UE_TRANSPORT/linear_preprocessing_rec.c
@@ -369,6 +369,6 @@ void compute_white_filter(float complex* H_re,
 
 float sqrt_float(float x, float sqrt_x)
 {
-  sqrt_x = (float)(sqrt((double)(x)));
+  sqrt_x = sqrtf(x);
   return sqrt_x;
 }
