diff --git a/openair1/PHY/NR_TRANSPORT/nr_ulsch_demodulation.c b/openair1/PHY/NR_TRANSPORT/nr_ulsch_demodulation.c
index 96266e279a..929bb0deac 100644
--- a/openair1/PHY/NR_TRANSPORT/nr_ulsch_demodulation.c
+++ b/openair1/PHY/NR_TRANSPORT/nr_ulsch_demodulation.c
@@ -1365,7 +1365,7 @@ static void inner_rx(PHY_VARS_gNB *gNB,
                             (int **)rxF_ch_maga, 
                             (int **)rxF_ch_magb, 
                             (int **)rxF_ch_magc, 
-                            (int **)chFext, 
+                            (int **)pusch_vars->ul_ch_estimates_ext, 
                             rel15_ul->rb_size,
                             frame_parms->nb_antennas_rx,
                             rel15_ul->qam_mod_order,
