diff --git a/openair1/PHY/INIT/nr_parms.c b/openair1/PHY/INIT/nr_parms.c
index 939478f35c..8a717e53ce 100644
--- a/openair1/PHY/INIT/nr_parms.c
+++ b/openair1/PHY/INIT/nr_parms.c
@@ -100,7 +100,7 @@ void set_Lmax(NR_DL_FRAME_PARMS *fp) {
 
 int nr_get_ssb_start_symbol(const NR_DL_FRAME_PARMS *fp, uint8_t i_ssb)
 {
-  int mu = fp->numerology_index;
+  uint8_t mu = fp->numerology_index;
   int symbol = 0;
   uint8_t n, n_temp;
   nr_ssb_type_e type = fp->ssb_type;
