diff --git a/openair2/RRC/NR/rrc_gNB_cuup.c b/openair2/RRC/NR/rrc_gNB_cuup.c
index 729f5f28b4..5c3be19bea 100644
--- a/openair2/RRC/NR/rrc_gNB_cuup.c
+++ b/openair2/RRC/NR/rrc_gNB_cuup.c
@@ -39,7 +39,7 @@ static const nr_rrc_cuup_container_t *select_cuup_slice(const struct rrc_cuup_tr
 {
   nr_rrc_cuup_container_t *second_best_match = NULL; /* if no NSSAI matches exactly */
   nr_rrc_cuup_container_t *cuup = NULL;
-  RB_FOREACH(cuup, rrc_cuup_tree, (struct rrc_cuup_tree *)&t) {
+  RB_FOREACH(cuup, rrc_cuup_tree, t) {
     e1ap_setup_req_t *sr = cuup->setup_req;
     for (int p = 0; p < sr->supported_plmns; ++p) {
       /* actually we should also check that the PLMN selected by the UE matches */
@@ -64,7 +64,7 @@ static const nr_rrc_cuup_container_t *select_cuup_round_robin(size_t n_t, const
    * ID % N with N number of CU-UPs */
   int m = (ue->rrc_ue_id - 1) % n_t;
   nr_rrc_cuup_container_t *cuup = NULL;
-  RB_FOREACH(cuup, rrc_cuup_tree, (struct rrc_cuup_tree *)&t) {
+  RB_FOREACH(cuup, rrc_cuup_tree, t) {
     if (m == 0) {
       LOG_W(RRC, "round-robin match: select CU-UP ID %ld (no NSSAI match)\n", cuup->setup_req->gNB_cu_up_id);
       return cuup;
