diff --git a/nfapi/open-nFAPI/vnf/inc/vnf_p7.h b/nfapi/open-nFAPI/vnf/inc/vnf_p7.h
index 4fcc99f79b..56f65e552d 100644
--- a/nfapi/open-nFAPI/vnf/inc/vnf_p7.h
+++ b/nfapi/open-nFAPI/vnf/inc/vnf_p7.h
@@ -88,9 +88,9 @@ typedef struct nfapi_vnf_p7_connection_info {
 	uint32_t previous_t2;
 	int32_t previous_sf_offset_filtered;
 	int32_t previous_slot_offset_filtered;
-	int sfn_sf;
-	int sfn;
-	int slot;
+	uint16_t sfn_sf;
+	uint16_t sfn;
+	uint16_t slot;
 
 	int socket;
 	struct sockaddr_in local_addr;
diff --git a/nfapi/open-nFAPI/vnf/src/vnf_p7.c b/nfapi/open-nFAPI/vnf/src/vnf_p7.c
index e4b4bef848..acac5073e1 100644
--- a/nfapi/open-nFAPI/vnf/src/vnf_p7.c
+++ b/nfapi/open-nFAPI/vnf/src/vnf_p7.c
@@ -422,7 +422,7 @@ int send_mac_slot_indications(vnf_p7_t* vnf_p7)
 	nfapi_vnf_p7_connection_info_t* curr = vnf_p7->p7_connections;
 	while(curr != 0)
 	{
-		if(curr->in_sync == 1)
+		if(curr->in_sync == 1u)
 		{
 			// ask for subframes in the future
 			//uint16_t sfn_sf_adv = increment_sfn_sf_by(curr->sfn_sf, 2);
@@ -444,7 +444,7 @@ int send_mac_subframe_indications(vnf_p7_t* vnf_p7)
 	nfapi_vnf_p7_connection_info_t* curr = vnf_p7->p7_connections;
 	while(curr != 0)
 	{
-		if(curr->in_sync == 1)
+		if(curr->in_sync == 1u)
 		{
 			// ask for subframes in the future
 			//uint16_t sfn_sf_adv = increment_sfn_sf_by(curr->sfn_sf, 2);
@@ -689,7 +689,7 @@ int vnf_nr_build_send_dl_node_sync(vnf_p7_t* vnf_p7, nfapi_vnf_p7_connection_inf
 int vnf_nr_sync(vnf_p7_t* vnf_p7, nfapi_vnf_p7_connection_info_t* p7_info)
 {
 
-	if(p7_info->in_sync == 1)
+	if(p7_info->in_sync == 1u)
 	{
 		uint16_t dl_sync_period_mask = p7_info->dl_in_sync_period-1;
 	//	uint16_t sfn_sf_dec = NFAPI_SFNSF2DEC(p7_info->sfn_sf);
@@ -720,7 +720,7 @@ int vnf_nr_sync(vnf_p7_t* vnf_p7, nfapi_vnf_p7_connection_info_t* p7_info)
 int vnf_sync(vnf_p7_t* vnf_p7, nfapi_vnf_p7_connection_info_t* p7_info)
 {
 
-	if(p7_info->in_sync == 1)
+	if(p7_info->in_sync == 1u)
 	{
 		uint16_t dl_sync_period_mask = p7_info->dl_in_sync_period-1;
 		uint16_t sfn_sf_dec = NFAPI_SFNSF2DEC(p7_info->sfn_sf);
@@ -1289,7 +1289,7 @@ void vnf_handle_ul_node_sync(void *pRecvMsg, int recvMsgLen, vnf_p7_t* vnf_p7)
 				{
 					if (phy->zero_count >= 10)
 					{
-						if(phy->in_sync == 0)
+						if(phy->in_sync == 0u)
 						{
 							//NFAPI_TRACE(NFAPI_TRACE_NOTE, "VNF P7 In Sync with phy (phy_id:%d)\n", phy->phy_id); 
 
@@ -1429,7 +1429,7 @@ void vnf_handle_ul_node_sync(void *pRecvMsg, int recvMsgLen, vnf_p7_t* vnf_p7)
 			}
 
 
-			if(phy->in_sync == 0)
+			if(phy->in_sync == 0u)
 			{
 				/*NFAPI_TRACE(NFAPI_TRACE_NOTE, "***** Adjusting VNF phy_id:%d SFN/SF (%s) from %d to %d (%d) mode:%s zeroCount:%u sync:%s\n",
 					ind.header.phy_id, (phy->in_sync ? "via sfn" : "now"),
