diff --git a/openair2/LAYER2/NR_MAC_UE/nr_ue_scheduler.c b/openair2/LAYER2/NR_MAC_UE/nr_ue_scheduler.c
index 7f7556b175..366f872a06 100644
--- a/openair2/LAYER2/NR_MAC_UE/nr_ue_scheduler.c
+++ b/openair2/LAYER2/NR_MAC_UE/nr_ue_scheduler.c
@@ -2322,9 +2322,9 @@ void nr_schedule_csi_for_im(NR_UE_MAC_INST_t *mac, int frame, int slot)
   }
 }
 
-NR_CSI_ResourceConfigId_t find_CSI_resourceconfig(NR_CSI_MeasConfig_t *csi_measconfig,
-                                                  NR_BWP_Id_t dl_bwp_id,
-                                                  NR_NZP_CSI_RS_ResourceId_t csi_id)
+int find_CSI_resourceconfig(NR_CSI_MeasConfig_t *csi_measconfig,
+                            NR_BWP_Id_t dl_bwp_id,
+                            NR_NZP_CSI_RS_ResourceId_t csi_id)
 {
   bool found = false;
   for (int csi_list = 0; csi_list < csi_measconfig->csi_ResourceConfigToAddModList->list.count; csi_list++) {
