diff --git a/openair2/M2AP/m2ap_MCE_interface_management.c b/openair2/M2AP/m2ap_MCE_interface_management.c
index 58a4c7b967..13bc0d08e4 100644
--- a/openair2/M2AP/m2ap_MCE_interface_management.c
+++ b/openair2/M2AP/m2ap_MCE_interface_management.c
@@ -1264,6 +1264,7 @@ int MCE_handle_ENB_CONFIGURATION_UPDATE(instance_t instance,
 
 int MCE_send_ENB_CONFIGURATION_UPDATE_FAILURE(instance_t instance,
                     m2ap_enb_configuration_update_failure_t * m2ap_enb_configuration_update_failure){
+  LOG_E(M2AP, "MCE_send_ENB_CONFIGURATION_UPDATE_FAILURE: Critical failure event\n");
   AssertFatal(1==0,"Not implemented yet\n");
 
 }
