diff --git a/openair2/F1AP/f1ap_du_ue_context_management.c b/openair2/F1AP/f1ap_du_ue_context_management.c
index 98de177af1..dc8595ae49 100644
--- a/openair2/F1AP/f1ap_du_ue_context_management.c
+++ b/openair2/F1AP/f1ap_du_ue_context_management.c
@@ -1668,7 +1668,7 @@ int DU_handle_UE_CONTEXT_MODIFICATION_REFUSE(instance_t instance, sctp_assoc_t a
                              container,
                              F1AP_ProtocolIE_ID_id_CriticalityDiagnostics,
                              false);
-  AssertFatal(ie == NULL, "handling of DRBs Modified List not implemented\n");
+  AssertFatal(ie == NULL, "handling of CriticalityDiagnostics not implemented\n");
 
   ue_context_modification_refuse(&refuse);
   return 0;
