diff --git a/openair2/LAYER2/nr_pdcp/cucp_cuup_handler.h b/openair2/LAYER2/nr_pdcp/cucp_cuup_handler.h
index 7290043371..5d6860f2a7 100644
--- a/openair2/LAYER2/nr_pdcp/cucp_cuup_handler.h
+++ b/openair2/LAYER2/nr_pdcp/cucp_cuup_handler.h
@@ -27,9 +27,10 @@
 void nr_pdcp_e1_if_init(bool uses_e1);
 
 struct e1ap_bearer_setup_req_s;
+struct e1ap_bearer_mod_req_s;
 struct e1ap_bearer_release_cmd_s;
 void e1_bearer_context_setup(const struct e1ap_bearer_setup_req_s *req);
-void e1_bearer_context_modif(const struct e1ap_bearer_setup_req_s *req);
+void e1_bearer_context_modif(const struct e1ap_bearer_mod_req_s *req);
 void e1_bearer_release_cmd(const struct e1ap_bearer_release_cmd_s *cmd);
 
 #endif /* CUCP_CUUP_HANDLER_H */
