diff --git a/openair2/COMMON/e1ap_messages_def.h b/openair2/COMMON/e1ap_messages_def.h
index 479aba28eb..5ed4a496f4 100644
--- a/openair2/COMMON/e1ap_messages_def.h
+++ b/openair2/COMMON/e1ap_messages_def.h
@@ -32,6 +32,10 @@ MESSAGE_DEF(E1AP_SETUP_REQ  , MESSAGE_PRIORITY_MED , e1ap_setup_req_t , e1ap_set
 MESSAGE_DEF(E1AP_SETUP_RESP , MESSAGE_PRIORITY_MED, e1ap_setup_resp_t , e1ap_setup_resp)
 /* GNB-CU-UP E1 Setup Failure: gNB-CU-CP -> gNB-CU-UP */
 MESSAGE_DEF(E1AP_SETUP_FAIL, MESSAGE_PRIORITY_MED, e1ap_setup_fail_t, e1ap_setup_fail)
+/* E1AP Trigger UE Reschedule: gNB-CU-CP -> RRC GNB */
+MESSAGE_DEF(E1AP_TRIGGER_UE_RESCHEDULE, MESSAGE_PRIORITY_MED, void, e1ap_trigger_ue_reschedule)
+
+
 
 /* E1AP Bearer Context Management Procedures */
 /* E1AP Bearer Context Setup Request: gNB-CU-CP -> gNB-CU-UP */
diff --git a/openair2/E1AP/e1ap.c b/openair2/E1AP/e1ap.c
index 1eff4ea529..e9afcd4445 100644
--- a/openair2/E1AP/e1ap.c
+++ b/openair2/E1AP/e1ap.c
@@ -176,6 +176,10 @@ static void e1apCUCP_send_SETUP_FAILURE(sctp_assoc_t assoc_id, const e1ap_setup_
   LOG_D(E1AP, "CU-CP: Encoding E1AP Setup Failure for transac_id %ld...\n", msg->transac_id);
   E1AP_E1AP_PDU_t *pdu = encode_e1ap_cuup_setup_failure(msg);
   e1ap_encode_send(CPtype, assoc_id, pdu, 0, __func__);
+
+  /* Trigger UE rescheduling to avoid missed scheduling opportunities for UEs with pending data */
+  MessageDef *sched_msg = itti_alloc_new_message(TASK_CUCP_E1, 0, E1AP_TRIGGER_UE_RESCHEDULE);
+  itti_send_msg_to_task(TASK_RRC_GNB, 0, sched_msg);
 }
 
 int e1apCUCP_handle_SETUP_REQUEST(sctp_assoc_t assoc_id, e1ap_upcp_inst_t *inst, const E1AP_E1AP_PDU_t *pdu)
