diff --git a/nfapi/open-nFAPI/pnf/src/pnf_p7_interface.c b/nfapi/open-nFAPI/pnf/src/pnf_p7_interface.c
index b021e67fc9..3f81664975 100644
--- a/nfapi/open-nFAPI/pnf/src/pnf_p7_interface.c
+++ b/nfapi/open-nFAPI/pnf/src/pnf_p7_interface.c
@@ -29,7 +29,7 @@ nfapi_pnf_p7_config_t* nfapi_pnf_p7_config_create()
 
 
 	// set the default parameters
-	_this->_public.segment_size = 65000; // UDP max packet size is 65535
+	_this->_public.segment_size = 1400; // Safe size to prevent uint16_t overflow
 	_this->max_num_segments = 8;
 	
 	_this->_public.subframe_buffer_size = 8;// TODO: Initialize the slot_buffer size
diff --git a/nfapi/open-nFAPI/vnf/src/vnf_p7_interface.c b/nfapi/open-nFAPI/vnf/src/vnf_p7_interface.c
index 225b0b7d8c..6ecc39586c 100644
--- a/nfapi/open-nFAPI/vnf/src/vnf_p7_interface.c
+++ b/nfapi/open-nFAPI/vnf/src/vnf_p7_interface.c
@@ -42,7 +42,7 @@ nfapi_vnf_p7_config_t* nfapi_vnf_p7_config_create()
 		return 0;
 
 	// todo : initialize
-	_this->_public.segment_size = 65000; // UDP max packet size is 65535
+	_this->_public.segment_size = 1400; // Safe size to prevent uint16_t overflow
 	_this->_public.max_num_segments = 8;
 	_this->_public.checksum_enabled = 1;
 
