diff --git a/openair3/NAS/UE/EMM/SAP/emm_send.c b/openair3/NAS/UE/EMM/SAP/emm_send.c
index 56708eded3..dbd206eb05 100644
--- a/openair3/NAS/UE/EMM/SAP/emm_send.c
+++ b/openair3/NAS/UE/EMM/SAP/emm_send.c
@@ -47,6 +47,7 @@ Description Defines functions executed at the EMMAS Service Access
 #include "emm_proc.h"
 
 #include <string.h> // strlen
+#include "OctetString.h"
 
 /****************************************************************************/
 /****************  E X T E R N A L    D E F I N I T I O N S  ****************/
@@ -776,7 +777,7 @@ int emm_send_authentication_response(const emm_as_security_t *msg,
 
   /* Mandatory - Authentication response parameter */
   size += AUTHENTICATION_RESPONSE_PARAMETER_MAXIMUM_LENGTH;
-  emm_msg->authenticationresponseparameter.res = *msg->res;
+  DUP_OCTET_STRING(*msg->res, emm_msg->authenticationresponseparameter.res);
 
   LOG_FUNC_RETURN (size);
 }
