diff --git a/openair2/LAYER2/nr_pdcp/nr_pdcp_ue_manager.c b/openair2/LAYER2/nr_pdcp/nr_pdcp_ue_manager.c
index 07edfb5814..0de0445f5e 100644
--- a/openair2/LAYER2/nr_pdcp/nr_pdcp_ue_manager.c
+++ b/openair2/LAYER2/nr_pdcp/nr_pdcp_ue_manager.c
@@ -134,7 +134,7 @@ void nr_pdcp_manager_remove_ue(nr_pdcp_ue_manager_t *_m, ue_id_t UEid)
     if (ue->srb[j] != NULL)
       ue->srb[j]->delete_entity(ue->srb[j]);
 
-  for (j = 0; j < 5; j++)
+  for (j = 0; j < MAX_DRBS_PER_UE; j++)
     if (ue->drb[j] != NULL)
       ue->drb[j]->delete_entity(ue->drb[j]);
 
@@ -178,7 +178,7 @@ void nr_pdcp_ue_add_srb_pdcp_entity(nr_pdcp_ue_t *ue, int srb_id, nr_pdcp_entity
 /* must be called with lock acquired */
 void nr_pdcp_ue_add_drb_pdcp_entity(nr_pdcp_ue_t *ue, int drb_id, nr_pdcp_entity_t *entity)
 {
-  if (drb_id < 1 || drb_id > 5) {
+  if (drb_id < 1 || drb_id > MAX_DRBS_PER_UE) {
     LOG_E(PDCP, "%s:%d:%s: fatal, bad drb id\n", __FILE__, __LINE__, __FUNCTION__);
     exit(1);
   }
