diff --git a/common/utils/tun_if.c b/common/utils/tun_if.c
index f74ab3c7f2..c037cfa532 100644
--- a/common/utils/tun_if.c
+++ b/common/utils/tun_if.c
@@ -239,7 +239,7 @@ bool tun_config(const char* ifname, const char *ipv4, const char *ipv6)
   return success;
 }
 
-void setup_ue_ipv4_route(const char* ifname, int instance_id, const char *ipv4)
+void setup_ue_ipv4_route(const char* ifname, ue_id_t instance_id, const char *ipv4)
 {
   int table_id = instance_id - 1 + 10000;
 
diff --git a/common/utils/tun_if.h b/common/utils/tun_if.h
index 8e1463624f..17319f05b2 100644
--- a/common/utils/tun_if.h
+++ b/common/utils/tun_if.h
@@ -24,6 +24,7 @@
 
 #include <stdbool.h>
 #include <net/if.h>
+#include "common/platform_types.h"
 
 /*!
  * \brief This function generates the name of the interface based on the prefix and
@@ -72,7 +73,7 @@ bool tun_config(const char* ifname, const char *ipv4, const char *ipv6);
  * \param[in] instance_id unique instance number, used to create the table
  * \param[in] ipv4 IPv4 address of the UE
  */
-void setup_ue_ipv4_route(const char* ifname, int instance_id, const char *ipv4);
+void setup_ue_ipv4_route(const char* ifname, ue_id_t instance_id, const char *ipv4);
 
 /*!
  * \brief This function allocates a TUN interface
