diff --git a/openair1/PHY/INIT/nr_parms.c b/openair1/PHY/INIT/nr_parms.c
index 2ace21eab5..662fd9611e 100644
--- a/openair1/PHY/INIT/nr_parms.c
+++ b/openair1/PHY/INIT/nr_parms.c
@@ -265,7 +265,7 @@ uint32_t get_slot_from_timestamp(openair0_timestamp timestamp_rx, const NR_DL_FR
    int samples_till_the_slot = fp->get_samples_per_slot(slot_idx,fp)-1;
    timestamp_rx = timestamp_rx%fp->samples_per_frame;
 
-    while (timestamp_rx > samples_till_the_slot) {
+    while (timestamp_rx >= samples_till_the_slot) {
         slot_idx++;
         samples_till_the_slot += fp->get_samples_per_slot(slot_idx,fp);
      }
