diff --git a/openair2/MCE_APP/mce_app.c b/openair2/MCE_APP/mce_app.c
index 53194d4899..2dd5f90070 100644
--- a/openair2/MCE_APP/mce_app.c
+++ b/openair2/MCE_APP/mce_app.c
@@ -76,7 +76,7 @@ static uint32_t MCE_app_register(uint32_t mce_id_start, uint32_t mce_id_end) {
     {
       // M3AP registration
         /* note:  there is an implicit relationship between the data structure and the message name */
-        msg_p = itti_alloc_new_message (TASK_MCE_APP, 0, M3AP_REGISTER_MCE_REQ);
+        msg_p = itti_alloc_new_message (TASK_MCE_APP, INSTANCE_DEFAULT, M3AP_REGISTER_MCE_REQ);
         //RCconfig_S1(msg_p, mce_id);
 
         //if (mce_id == 0) 
@@ -157,7 +157,7 @@ static uint32_t MCE_app_register_m3(uint32_t mce_id_start, uint32_t mce_id_end)
   for (mce_id = mce_id_start; (mce_id < mce_id_end) ; mce_id++) {
     {
   //	LOG_W(MCE_APP,"Register commes inside ...\n");
-      msg_p = itti_alloc_new_message (TASK_MCE_APP, 0, M3AP_REGISTER_MCE_REQ);
+      msg_p = itti_alloc_new_message (TASK_MCE_APP, INSTANCE_DEFAULT, M3AP_REGISTER_MCE_REQ);
       RCconfig_M3(msg_p, mce_id);
       itti_send_msg_to_task (TASK_M3AP_MCE, ENB_MODULE_ID_TO_INSTANCE(mce_id), msg_p);
   	LOG_D(MCE_APP,"Register sent ...\n");
@@ -172,7 +172,7 @@ static uint32_t MCE_app_register_m3(uint32_t mce_id_start, uint32_t mce_id_end)
 //static uint32_t MCE_app_handle_m3ap_mbms_session_start_req(instance_t instance){
 //  	//uint32_t         mce_id=0;
 //  //	MessageDef      *msg_p;
-//  //      msg_p = itti_alloc_new_message (TASK_MCE_APP, 0, M3AP_MBMS_SESSION_START_RESP);
+//  //      msg_p = itti_alloc_new_message (TASK_MCE_APP, INSTANCE_DEFAULT, M3AP_MBMS_SESSION_START_RESP);
 //  //      itti_send_msg_to_task (TASK_M3AP_MCE, ENB_MODULE_ID_TO_INSTANCE(instance), msg_p);
 //	
 //	
@@ -182,7 +182,7 @@ static uint32_t MCE_app_register_m3(uint32_t mce_id_start, uint32_t mce_id_end)
 static uint32_t MCE_app_handle_m3ap_mbms_session_stop_req(instance_t instance){
   	//uint32_t         mce_id=0;
   	MessageDef      *msg_p;
-        msg_p = itti_alloc_new_message (TASK_MCE_APP, 0, M3AP_MBMS_SESSION_STOP_RESP);
+        msg_p = itti_alloc_new_message (TASK_MCE_APP, INSTANCE_DEFAULT, M3AP_MBMS_SESSION_STOP_RESP);
         itti_send_msg_to_task (TASK_M3AP_MCE, ENB_MODULE_ID_TO_INSTANCE(instance), msg_p);
 	
 	return 0;
@@ -191,7 +191,7 @@ static uint32_t MCE_app_handle_m3ap_mbms_session_stop_req(instance_t instance){
 static uint32_t MCE_app_handle_m3ap_mbms_session_update_req(instance_t instance){
   	//uint32_t         mce_id=0;
   	MessageDef      *msg_p;
-        msg_p = itti_alloc_new_message (TASK_MCE_APP, 0, M3AP_MBMS_SESSION_UPDATE_RESP);
+        msg_p = itti_alloc_new_message (TASK_MCE_APP, INSTANCE_DEFAULT, M3AP_MBMS_SESSION_UPDATE_RESP);
         itti_send_msg_to_task (TASK_M3AP_MCE, ENB_MODULE_ID_TO_INSTANCE(instance), msg_p);
 	
 	return 0;
@@ -204,7 +204,7 @@ static uint32_t MCE_app_handle_m2ap_setup_req(instance_t instance){
 	
   	//uint32_t         mce_id=0;
   	MessageDef      *msg_p;
-        msg_p = itti_alloc_new_message (TASK_MCE_APP, 0, M2AP_SETUP_RESP);
+        msg_p = itti_alloc_new_message (TASK_MCE_APP, INSTANCE_DEFAULT, M2AP_SETUP_RESP);
 	if(m2ap_setup_resp_local)
 		memcpy(&M2AP_SETUP_RESP(msg_p),m2ap_setup_resp_local,sizeof(m2ap_setup_resp_t));
 	else
@@ -218,7 +218,7 @@ static uint32_t MCE_app_handle_m2ap_setup_req(instance_t instance){
 
 static uint32_t MCE_app_handle_m2ap_mbms_session_start_resp(instance_t instance){
 	MessageDef      *msg_p;
-  	msg_p = itti_alloc_new_message (TASK_MCE_APP, 0, M3AP_MBMS_SESSION_START_RESP); 
+  	msg_p = itti_alloc_new_message (TASK_MCE_APP, INSTANCE_DEFAULT, M3AP_MBMS_SESSION_START_RESP); 
 	itti_send_msg_to_task (TASK_M3AP_MCE, ENB_MODULE_ID_TO_INSTANCE(instance), msg_p);
 	return 0;
 }
@@ -231,7 +231,7 @@ static uint32_t MCE_app_send_m2ap_mbms_scheduling_information(instance_t instanc
 	
   	//uint32_t         mce_id=0;
   	MessageDef      *msg_p;
-        msg_p = itti_alloc_new_message (TASK_MCE_APP, 0, M2AP_MBMS_SCHEDULING_INFORMATION);
+        msg_p = itti_alloc_new_message (TASK_MCE_APP, INSTANCE_DEFAULT, M2AP_MBMS_SCHEDULING_INFORMATION);
 	if(m2ap_mbms_scheduling_information_local)
 		memcpy(&M2AP_MBMS_SCHEDULING_INFORMATION(msg_p),m2ap_mbms_scheduling_information_local,sizeof(m2ap_mbms_scheduling_information_t));
 	else
@@ -245,7 +245,7 @@ static uint32_t MCE_app_send_m2ap_session_start_req(instance_t instance){
 	
   	//uint32_t         mce_id=0;
   	MessageDef      *msg_p;
-        msg_p = itti_alloc_new_message (TASK_MCE_APP, 0, M2AP_MBMS_SESSION_START_REQ);
+        msg_p = itti_alloc_new_message (TASK_MCE_APP, INSTANCE_DEFAULT, M2AP_MBMS_SESSION_START_REQ);
         itti_send_msg_to_task (TASK_M2AP_MCE, ENB_MODULE_ID_TO_INSTANCE(instance), msg_p);
 	
 	return 0;
