diff --git a/openair2/LAYER2/MAC/eNB_scheduler_ulsch.c b/openair2/LAYER2/MAC/eNB_scheduler_ulsch.c
index 058ce4db34..5c2f9e4b3f 100644
--- a/openair2/LAYER2/MAC/eNB_scheduler_ulsch.c
+++ b/openair2/LAYER2/MAC/eNB_scheduler_ulsch.c
@@ -516,8 +516,7 @@ rx_sdu(const module_id_t enb_mod_idP,
                                             DCCH,
                                             (uint8_t *) payload_ptr,
                                             rx_lengths[i],
-                                            0,
-                                            ra->rach_resource_type > 0
+                                            0
                                            );
 
               /* Received a new rnti */
@@ -781,7 +780,7 @@ rx_sdu(const module_id_t enb_mod_idP,
                            CCCH,
                            (uint8_t *) payload_ptr,
                            rx_lengths[i],
-                           0,ra->rach_resource_type > 0
+                           0
                           );
 
           if (num_ce > 0) { // handle msg3 which is not RRCConnectionRequest
