diff --git a/openair2/LAYER2/NR_MAC_gNB/nr_mac_gNB.h b/openair2/LAYER2/NR_MAC_gNB/nr_mac_gNB.h
index 2627c9162b..f4cb22d5e8 100644
--- a/openair2/LAYER2/NR_MAC_gNB/nr_mac_gNB.h
+++ b/openair2/LAYER2/NR_MAC_gNB/nr_mac_gNB.h
@@ -785,8 +785,8 @@ typedef struct {
   /// list of allocated beams per period
   int **beam_allocation;
   int beam_duration; // in slots
-  int beams_per_period;
-  int beam_allocation_size;
+  unsigned int beams_per_period;
+  unsigned int beam_allocation_size;
   nr_beam_mode_t beam_mode;
 } NR_beam_info_t;
 
