diff --git a/nfapi/open-nFAPI/nfapi/public_inc/nfapi_interface.h b/nfapi/open-nFAPI/nfapi/public_inc/nfapi_interface.h
index 3312fa832e..0d9d9de996 100644
--- a/nfapi/open-nFAPI/nfapi/public_inc/nfapi_interface.h
+++ b/nfapi/open-nFAPI/nfapi/public_inc/nfapi_interface.h
@@ -70,6 +70,7 @@
 
 #define NFAPI_HEADER_LENGTH 8
 #define NFAPI_P7_HEADER_LENGTH 16
+#define NFAPI_PDU_TYPE_FIELD_SIZE 1
 
 #define NFAPI_VENDOR_EXTENSION_MIN_TAG_VALUE 0xF000
 #define NFAPI_VENDOR_EXTENSION_MAX_TAG_VALUE 0xFFFF
diff --git a/nfapi/open-nFAPI/nfapi/src/nfapi_p7.c b/nfapi/open-nFAPI/nfapi/src/nfapi_p7.c
index bcaf7a524d..0190d30756 100644
--- a/nfapi/open-nFAPI/nfapi/src/nfapi_p7.c
+++ b/nfapi/open-nFAPI/nfapi/src/nfapi_p7.c
@@ -822,7 +822,7 @@ static uint8_t pack_dl_config_request_body_value(void *tlv, uint8_t **ppWritePac
     };
 
     // add 1 for the pdu_type. The delta will include the pdu_size
-    pdu->pdu_size = 1 + (*ppWritePackedMsg - pWritePackedMsgPduSize);
+    pdu->pdu_size = NFAPI_PDU_TYPE_FIELD_SIZE + (*ppWritePackedMsg - pWritePackedMsgPduSize);
 
     push8(pdu->pdu_size, &pWritePackedMsgPduSize, end);
   }
@@ -1647,7 +1647,7 @@ static uint8_t pack_ul_config_request_body_value(void *tlv, uint8_t **ppWritePac
     };
 
     // add 1 for the pdu_type. The delta will include the pdu_size
-    pdu->pdu_size = 1 + (*ppWritePackedMsg - pWritePackedMsgPduSize);
+    pdu->pdu_size = NFAPI_PDU_TYPE_FIELD_SIZE + (*ppWritePackedMsg - pWritePackedMsgPduSize);
 
     push8(pdu->pdu_size, &pWritePackedMsgPduSize, end);
   }
@@ -1893,7 +1893,7 @@ static uint8_t pack_hi_dci0_request_body_value(void *tlv, uint8_t **ppWritePacke
     };
 
     // add 1 for the pdu_type. The delta will include the pdu_size
-    pdu->pdu_size = 1 + (*ppWritePackedMsg - pWritePackedMsgPduSize);
+    pdu->pdu_size = NFAPI_PDU_TYPE_FIELD_SIZE + (*ppWritePackedMsg - pWritePackedMsgPduSize);
 
     push8(pdu->pdu_size, &pWritePackedMsgPduSize, end);
   }
@@ -2913,7 +2913,7 @@ static uint8_t pack_lbt_dl_config_request_body_value(void *tlv, uint8_t **ppWrit
     };
 
     // add 1 for the pdu_type. The delta will include the pdu_size
-    pdu->pdu_size = 1 + (*ppWritePackedMsg - pWritePackedMsgPduSize);
+    pdu->pdu_size = NFAPI_PDU_TYPE_FIELD_SIZE + (*ppWritePackedMsg - pWritePackedMsgPduSize);
 
     push8(pdu->pdu_size, &pWritePackedMsgPduSize, end);
   }
@@ -2985,7 +2985,7 @@ static uint8_t pack_lbt_dl_config_indication_value(void *tlv, uint8_t **ppWriteP
     };
 
     // add 1 for the pdu_type. The delta will include the pdu_size
-    pdu->pdu_size = 1 + (*ppWritePackedMsg - pWritePackedMsgPduSize);
+    pdu->pdu_size = NFAPI_PDU_TYPE_FIELD_SIZE + (*ppWritePackedMsg - pWritePackedMsgPduSize);
 
     push8(pdu->pdu_size, &pWritePackedMsgPduSize, end);
   }
