diff --git a/openair2/RRC/NR/nr_rrc_config.c b/openair2/RRC/NR/nr_rrc_config.c
index 9bbd4f38b7..13e83e404d 100644
--- a/openair2/RRC/NR/nr_rrc_config.c
+++ b/openair2/RRC/NR/nr_rrc_config.c
@@ -588,63 +588,63 @@ static struct NR_SRS_Resource__resourceType__periodic *configure_periodic_srs(co
   struct NR_SRS_Resource__resourceType__periodic *periodic_srs = calloc(1,sizeof(*periodic_srs));
   if (ideal_period < 5) {
     periodic_srs->periodicityAndOffset_p.present = NR_SRS_PeriodicityAndOffset_PR_sl4;
-    periodic_srs->periodicityAndOffset_p.choice.sl4 = offset;
+    periodic_srs->periodicityAndOffset_p.choice.sl4 = offset % 4;
   }
   else if (ideal_period < 6) {
     periodic_srs->periodicityAndOffset_p.present = NR_SRS_PeriodicityAndOffset_PR_sl5;
-    periodic_srs->periodicityAndOffset_p.choice.sl5 = offset;
+    periodic_srs->periodicityAndOffset_p.choice.sl5 = offset % 5;
   }
   else if (ideal_period < 9) {
     periodic_srs->periodicityAndOffset_p.present = NR_SRS_PeriodicityAndOffset_PR_sl8;
-    periodic_srs->periodicityAndOffset_p.choice.sl8 = offset;
+    periodic_srs->periodicityAndOffset_p.choice.sl8 = offset % 8;
   }
   else if (ideal_period < 11) {
     periodic_srs->periodicityAndOffset_p.present = NR_SRS_PeriodicityAndOffset_PR_sl10;
-    periodic_srs->periodicityAndOffset_p.choice.sl10 = offset;
+    periodic_srs->periodicityAndOffset_p.choice.sl10 = offset % 10;
   }
   else if (ideal_period < 17) {
     periodic_srs->periodicityAndOffset_p.present = NR_SRS_PeriodicityAndOffset_PR_sl16;
-    periodic_srs->periodicityAndOffset_p.choice.sl16 = offset;
+    periodic_srs->periodicityAndOffset_p.choice.sl16 = offset % 16;
   }
   else if (ideal_period < 21) {
     periodic_srs->periodicityAndOffset_p.present = NR_SRS_PeriodicityAndOffset_PR_sl20;
-    periodic_srs->periodicityAndOffset_p.choice.sl20 = offset;
+    periodic_srs->periodicityAndOffset_p.choice.sl20 = offset % 20;
   }
   else if (ideal_period < 33) {
     periodic_srs->periodicityAndOffset_p.present = NR_SRS_PeriodicityAndOffset_PR_sl32;
-    periodic_srs->periodicityAndOffset_p.choice.sl32 = offset;
+    periodic_srs->periodicityAndOffset_p.choice.sl32 = offset % 32;
   }
   else if (ideal_period < 41) {
     periodic_srs->periodicityAndOffset_p.present = NR_SRS_PeriodicityAndOffset_PR_sl40;
-    periodic_srs->periodicityAndOffset_p.choice.sl40 = offset;
+    periodic_srs->periodicityAndOffset_p.choice.sl40 = offset % 40;
   }
   else if (ideal_period < 65) {
     periodic_srs->periodicityAndOffset_p.present = NR_SRS_PeriodicityAndOffset_PR_sl64;
-    periodic_srs->periodicityAndOffset_p.choice.sl64 = offset;
+    periodic_srs->periodicityAndOffset_p.choice.sl64 = offset % 64;
   }
   else if (ideal_period < 81) {
     periodic_srs->periodicityAndOffset_p.present = NR_SRS_PeriodicityAndOffset_PR_sl80;
-    periodic_srs->periodicityAndOffset_p.choice.sl80 = offset;
+    periodic_srs->periodicityAndOffset_p.choice.sl80 = offset % 80;
   }
   else if (ideal_period < 161) {
     periodic_srs->periodicityAndOffset_p.present = NR_SRS_PeriodicityAndOffset_PR_sl160;
-    periodic_srs->periodicityAndOffset_p.choice.sl160 = offset;
+    periodic_srs->periodicityAndOffset_p.choice.sl160 = offset % 160;
   }
   else if (ideal_period < 321) {
     periodic_srs->periodicityAndOffset_p.present = NR_SRS_PeriodicityAndOffset_PR_sl320;
-    periodic_srs->periodicityAndOffset_p.choice.sl320 = offset;
+    periodic_srs->periodicityAndOffset_p.choice.sl320 = offset % 320;
   }
   else if (ideal_period < 641) {
     periodic_srs->periodicityAndOffset_p.present = NR_SRS_PeriodicityAndOffset_PR_sl640;
-    periodic_srs->periodicityAndOffset_p.choice.sl640 = offset;
+    periodic_srs->periodicityAndOffset_p.choice.sl640 = offset % 640;
   }
   else if (ideal_period < 1281) {
     periodic_srs->periodicityAndOffset_p.present = NR_SRS_PeriodicityAndOffset_PR_sl1280;
-    periodic_srs->periodicityAndOffset_p.choice.sl1280 = offset;
+    periodic_srs->periodicityAndOffset_p.choice.sl1280 = offset % 1280;
   }
   else {
     periodic_srs->periodicityAndOffset_p.present = NR_SRS_PeriodicityAndOffset_PR_sl2560;
-    periodic_srs->periodicityAndOffset_p.choice.sl2560 = offset;
+    periodic_srs->periodicityAndOffset_p.choice.sl2560 = offset % 2560;
   }
   return periodic_srs;
 }
