diff --git a/openair2/E1AP/e1ap.c b/openair2/E1AP/e1ap.c
index 1eff4ea529..b79786b305 100644
--- a/openair2/E1AP/e1ap.c
+++ b/openair2/E1AP/e1ap.c
@@ -116,8 +116,12 @@ static int e1ap_handle_message(instance_t instance, sctp_assoc_t assoc_id, const
 static void e1_task_handle_sctp_data_ind(instance_t instance, sctp_data_ind_t *sctp_data_ind)
 {
   int result;
+  int ret;
   DevAssert(sctp_data_ind != NULL);
-  e1ap_handle_message(instance, sctp_data_ind->assoc_id, sctp_data_ind->buffer, sctp_data_ind->buffer_length);
+  ret = e1ap_handle_message(instance, sctp_data_ind->assoc_id, sctp_data_ind->buffer, sctp_data_ind->buffer_length);
+  if (ret < 0) {
+    LOG_E(E1AP, "e1ap_handle_message failed with error %d\n", ret);
+  }
   result = itti_free(TASK_UNKNOWN, sctp_data_ind->buffer);
   AssertFatal(result == EXIT_SUCCESS, "Failed to free memory (%d)!\n", result);
 }
