diff --git a/openair1/PHY/LTE_TRANSPORT/edci.c b/openair1/PHY/LTE_TRANSPORT/edci.c
index f3a43584fb..a6ba0e6743 100644
--- a/openair1/PHY/LTE_TRANSPORT/edci.c
+++ b/openair1/PHY/LTE_TRANSPORT/edci.c
@@ -133,7 +133,7 @@ void init_mpdcch5ss2tab_normal_regular_subframe_evenNRBDL(PHY_VARS_eNB *eNB) {
   // NOTE : THIS IS FOR TM1 ONLY FOR NOW!!!!!!!
   LOG_D(PHY, "Inititalizing mpdcch5ss2tab for normal prefix, normal prefix, no PSS/SSS/PBCH, even N_RB_DL\n");
 
-  for (l = 2; l < 14; l++) {
+  for (l = 2; l < 13; l++) {
     for (k = 0; k < 72; k++) {
       kmod = k % 12;
 
