diff --git a/openair3/SECU/snow3g.c b/openair3/SECU/snow3g.c
index 2278a6fe94..7a877043ba 100644
--- a/openair3/SECU/snow3g.c
+++ b/openair3/SECU/snow3g.c
@@ -671,7 +671,7 @@ static uint64_t MULxPOW64(uint64_t v, int i, uint64_t c)
 static uint64_t MUL64(uint64_t v, uint64_t p, uint64_t c)
 {
   uint64_t result = 0;
-  for (int i = 0; i < 64; i++) {
+  for (int i = 0; i < 63; i++) {
     if ((p >> i) & 1)
       result ^= MULxPOW64(v, i, c);
   }
@@ -759,7 +759,7 @@ static void _snow3g_integrity(uint32_t count, uint32_t fresh, int direction, con
   pm[0][0] = pm[1][0] = pm[2][0] = pm[3][0] = pm[4][0] = pm[5][0] = pm[6][0] = pm[7][0] = 0;
   pm[0][1] = p;
 
-  for (int i = 1; i <= 63; i++) {
+  for (int i = 1; i <= 62; i++) {
     pm[i >> 3][1 << (i & 0x07)] = pm[(i - 1) >> 3][1 << ((i - 1) & 0x07)] << 1;
     if (pm[(i - 1) >> 3][1 << ((i - 1) & 0x07)] & 0x8000000000000000ULL)
       pm[i >> 3][1 << (i & 0x07)] = pm[i >> 3][1 << (i & 0x07)] ^ 0x1b;
